pkg/cloudprovider/aws/aws.go

```diff
@@ -119,14 +119,14 @@ func getAuth() (auth aws.Auth, err error) {
 
 // readAWSCloudConfig reads an instance of AWSCloudConfig from config reader.
 func readAWSCloudConfig(config io.Reader, metadata AWSMetadata) (*AWSCloudConfig, error) {
-	if config == nil {
```

Comment:
without this check, if `config == nil`, wouldn't we then be returning `&nil` at line 145 below?